tamr_unify_client/attribute/subattribute.py

```diff
@@ -1,43 +1,38 @@
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import Any, Dict, Optional
+
+from tamr_unify_client.attribute.type import AttributeType
+
+SubAttributeJson = Dict[str, Any]
+
+@dataclass(frozen=True)
 class SubAttribute:
-    """
-    An attribute which is itself a property of another attribute.
+    """An attribute which is itself a property of another attribute.
 
     See https://docs.tamr.com/reference#attribute-types
 
-    :param data: JSON data representing this attribute
-    :type data: :py:class:`dict`
+    Args:
+        name: Name of sub-attribute
+        description: Description of sub-attribute
+        type: See https://docs.tamr.com/reference#attribute-types
+        is_nullable: If this sub-attribute can be null
     """
+    name: str
+    type: AttributeType
+    is_nullable: bool
+    _json: SubAttributeJson = field(repr=False)
+    description: Optional[str] = None
+
+    @staticmethod
+    def from_json(d: SubAttributeJson) -> 'SubAttribute':
```

Comment:
In Python 3.6, cyclic type references (e.g. the return type of this function refers to the type of the class implementing it) require using strings. See [PEP 563](https://www.python.org/dev/peps/pep-0563/) for how this can be made easier in Python 3.7

Reply:
what's the policy for upgrading/supporting python versions?

Reply:
we support 3.6+ for Python Client. Ubuntu 19 and many modern linux distros ship with 3.6, so this decision keeps life simple for DataOps folks @ Tamr